### PR TITLE
Fix ahnyQuab ogg decompression crashes.

### LINKED ISSUE #1506: Patcher crashes on OGG files
Patching is sometimes crashing. To reproduce:

- Delete all files in an Uru installation except for the server.ini.
- Run plFilePatcher (or the normal game client) on that directory.
- The game will be redownloaded either completely or partially
- Whichever client is used will throw exceptions within plOGGCodec for various OGG files

It's not clear if this is a server issue or a client issue, but I'm leaning client issue because:
- The build from August 2022 that was shared with Apple DTS is still ok.
- These OGG files have not changed.
- The MD5 hash of the OGG files matches existing downloaded installations.
- The OGG files playback on disk.

This bug was reproduced in ca80c0d8c16daf291ead344de168012b0c0ed259 - and I do not have a good commit hash that works so far.

### PR DESCRIPTION
**Pull request by Hoikas**
This issue was introduced when `hsStream::Close()` was removed in favor of RAII in #1448. Unfortunately, the patcher is handing off files before it deletes the patcher stream, which causes race conditions around zlib decompression not being complete. We need to explicitly close the underlying stream before performing any handoffs to prevent crashes.

Fixes #1506

### CODE REVIEW

File: Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp [hunk 1]
```diff
@@ -264,6 +264,13 @@ class pfPatcherStream : public plZlibStream
         return retVal;
     }
 
+    void Close()
```
Comment (dpogue) [inline Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp:267]: is this `override`?

Comment (Hoikas) [reply Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp:267]: There is no more `virtual Close();` as of #1448.

Comment (dpogue) [reply Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp:267]: Ah okay

**Review by dpogue — APPROVED** [review]

**Review by Deledrius — APPROVED** [review]

**Comment by dpogue** [comment]
Can confirm that this allowed me to patch the entire client data set from TrollLand with no issues.

**Pull request merged**